File: python_django/core_test/customer/domain/customer_mother.py
from core.customer.domain.customer import Customer
from core_test.customer.domain.customer_id_mother import CustomerIdMother
from core_test.customer.domain.customer_first_name_mother import CustomerFirstNameMother


class CustomerMother:
    @staticmethod
    def random() -> Customer:
        customer_id = CustomerIdMother.random()
        customer_name = CustomerFirstNameMother.random()

        return Customer(customer_id, customer_name)

File: typescript_nestjs/test/core/customer/domain/CustomerMother.ts
import {CustomerId} from "../../../../src/core/customer/domain/CustomerId";
import {Customer} from "../../../../src/core/customer/domain/Customer";
import {CustomerIdentificationMother} from "./CustomerIdentificationMother";
import {CustomerFirstNameMother} from "./CustomerFirstNameMother";
import {CustomerLastNameMother} from "./CustomerLastNameMother";
import {CustomerMobilePhoneMother} from "./CustomerMobilePhoneMother";
import {CustomerIdMother} from "./CustomerIdMother";

export class CustomerMother {
    static withCustomerId(customerId: CustomerId): Customer {
        return new Customer(
            customerId,
            CustomerIdentificationMother.random(),
            CustomerFirstNameMother.random(),
            CustomerLastNameMother.random(),
            CustomerMobilePhoneMother.random()
        );
    }

    static random() {
        return new Customer(
            CustomerIdMother.random(),
            CustomerIdentificationMother.random(),
            CustomerFirstNameMother.random(),
            CustomerLastNameMother.random(),
            CustomerMobilePhoneMother.random()
        );
    }
}
